src/Snackbar/Snackbar.react.js

```diff
@@ -0,0 +1,207 @@
+            props.style.actionText,
+        ],
+    };
+}
+
+/**
+* Component for snackbars
+* https://material.io/guidelines/components/snackbars-toasts.html
+*/
+class Snackbar extends PureComponent {
+    constructor(props, context) {
+        super(props, context);
+        const styles = getStyles(props, context);
+        this.state = {
+            styles,
+            moveAnimated: new Animated.Value(StyleSheet.flatten(styles.container).height),
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.visible !== this.props.visible) {
+            if (nextProps.visible === true) {
+                this.show(nextProps.bottomNavigation);
+                this.setHideTimer();
+            } else {
+                this.hide();
+            }
+        } else if ((nextProps.bottomNavigation !== this.props.bottomNavigation)
+        && nextProps.visible) {
+            this.move(nextProps.bottomNavigation);
+        }
+    }
+
+    componentWillUnmount() {
+        clearTimeout(this.hideTimer);
+    }
+
+    setHideTimer() {
+        const { timeout, onRequestClose } = this.props;
+
+        if (timeout > 0) {
+            clearTimeout(this.hideTimer);
+            this.hideTimer = setTimeout(() => {
+                onRequestClose();
+            }, timeout);
+        }
+    }
+
+    show = (bottomNavigation) => {
+        let toValue = 0;
+        if (bottomNavigation) {
+            // TODO: Get bottom navigation height from context.
+            toValue = -56;
+        }
+
+        Animated.timing(this.state.moveAnimated, {
+            toValue,
+            duration: 225,
+            easing: Easing.bezier(0.0, 0.0, 0.2, 1),
+            useNativeDriver: Platform.OS === 'android',
+        }).start();
+    }
+
+    hide = () => {
+        const { moveAnimated, styles } = this.state;
+        Animated.timing(moveAnimated, {
+            toValue: (StyleSheet.flatten(styles.container).height),
+            duration: 195,
+            easing: Easing.bezier(0.4, 0.0, 1, 1),
+            useNativeDriver: Platform.OS === 'android',
+        }).start();
+    }
+
+    move = (bottomNavigation) => {
+        const { moveAnimated } = this.state;
+        const toValue = bottomNavigation ? -56 : 0;
+        const duration = bottomNavigation ? 225 : 195;
+        const easing = bottomNavigation ?
+          Easing.bezier(0.0, 0.0, 0.2, 1) : Easing.bezier(0.4, 0.0, 0.6, 1);
+
+        Animated.timing(moveAnimated, {
+            toValue,
+            duration,
+            easing,
+            useNativeDriver: Platform.OS === 'android',
+        }).start();
+    }
+
+    renderAction = () => {
+        const { actionText, actionHandler } = this.props;
+        const { styles } = this.state;
+
+        if (actionText && (typeof actionHandler === 'function')) {
+            return (
+                <Button
+                    style={{ container: styles.actionContainer, text: styles.actionText }}
```

Comment:
`style={styles.button}` ... unfortunately, there is no way how to change the button text I think .. :/ But it's an issue in button component, not here ..